packages/botonic-nlp/src/embeddings/database/downloader.ts

```diff
@@ -0,0 +1,53 @@
+import axios from 'axios'
+import { createWriteStream, existsSync, mkdirSync } from 'fs'
+import { join } from 'path'
+
+import { Locale } from '../../types'
+import { EMBEDDINGS_URL, GLOBAL_EMBEDDINGS_PATH } from './constants'
+import { EmbeddingsDimension, EmbeddingsType } from './types'
+
+export class DatabaseDownloader {
+  static async download(
+    locale: Locale,
+    type: EmbeddingsType,
+    dimension: EmbeddingsDimension
+  ): Promise<void> {
+    const filename = `${type}-${dimension}d-${locale}.db`
+    this.createDirectory()
+    await this.downloadIntoPath(
+      `${EMBEDDINGS_URL}/${filename}`,
+      join(GLOBAL_EMBEDDINGS_PATH, filename)
+    )
+  }
```

Comment:
this logic should be done in the manager, who calls downloadIntoPath with the already generated name for the file which we want to download.